### PR TITLE
Release 4.0.01

### PR DESCRIPTION
**Pull request by ndellingwood**
Patch release

### CODE REVIEW

**Comment by ndellingwood** [comment]
Please hold merge until successful testing of https://github.com/trilinos/Trilinos/pull/11817

**Comment by ndellingwood** [comment]
The CI GCC-8.4.0 build had I believe a spurious test failure in `KokkosContainers_UnitTest_OpenMP`:
```
25: [ RUN      ] openmp.scatterview
25/33 Test #25: KokkosContainers_UnitTest_OpenMP ............Child killed***Exception:  30.31 sec
```

Possibly relevant #6059

File: CMakeLists.txt [hunk 1]
```diff
@@ -139,7 +144,7 @@ ENDIF()
 
 set(Kokkos_VERSION_MAJOR 4)
 set(Kokkos_VERSION_MINOR 0)
-set(Kokkos_VERSION_PATCH 0)
+set(Kokkos_VERSION_PATCH 01)
```
Comment (dalg24) [inline CMakeLists.txt:147]:
```suggestion
set(Kokkos_VERSION_PATCH 1)
```

Comment (ndellingwood) [reply CMakeLists.txt]: @dalg24 fixed with [0ab1f11](https://github.com/kokkos/kokkos/pull/6080/commits/0ab1f11fe31ec7e180537252f4732a6c9573b27e)

**Review by dalg24 — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (ndellingwood) [reply CMakeLists.txt]: I should have mentioned the snapshot to Trilinos is also updated with the corrected patch version convention

**Comment by ndellingwood** [comment]
https://github.com/trilinos/Trilinos/pull/11817 has merged, no longer a blocker

**Review by crtrott — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -102,6 +121,7 @@
 - Add missing `ReductionIdentity<char>` specialization [\#5798](https://github.com/kokkos/kokkos/pull/5798)
 - Don't install standard algorithms headers multiple times [\#5670](https://github.com/kokkos/kokkos/pull/5670)
 - Fix max scratch size calculation for level 0 scratch in CUDA and HIP [\#5718](https://github.com/kokkos/kokkos/pull/5718)
+- Fix excessive build times using Makefile.kokkos [\#6068](https://github.com/kokkos/kokkos/pull/6068)
```
Comment (dalg24) [inline CHANGELOG.md:124]: Please fix this entry in develop when you cherry-pitch the changelog

**Pull request merged**